app/controllers/application_controller.rb

```diff
@@ -227,6 +227,7 @@ def fix_broken_personal_key_url
   end
 
   def after_sign_in_path_for(_user)
+    return user_password_compromised_url if session[:redirect_to_password_compromised].present?
     return rules_of_use_path if !current_user.accepted_rules_of_use_still_valid?
     return user_please_call_url if current_user.suspended?
```

Comment:
Minor: I think Rules of Use and suspended considerations should probably take priority.

```suggestion
    return rules_of_use_path if !current_user.accepted_rules_of_use_still_valid?
    return user_please_call_url if current_user.suspended?
    return user_password_compromised_url if session[:redirect_to_password_compromised].present?
```